src/Microsoft.OData.Edm/Schema/EdmAnnotationsTarget.cs

```diff
@@ -0,0 +1,55 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmAnnotationsTarget.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Edm.Vocabularies;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Represents an EDM annotations target.
+    /// </summary>
+    public class EdmAnnotationsTarget : IEdmAnnotationsTarget
+    {
+        private IEnumerable<IEdmElement> targetSegments;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmAnnotationsTarget"/> class.
+        /// </summary>
+        /// <param name="targetSegments">Target segments.</param>
+        public EdmAnnotationsTarget(IEnumerable<IEdmElement> targetSegments)
+        {
+            EdmUtil.CheckArgumentNull(targetSegments, nameof(targetSegments));
+
+            // Validate that the first element is Container.
+            if (!(targetSegments.First() is IEdmEntityContainer))
+            {
+                throw new InvalidOperationException(Strings.EdmAnnotationsTarget_FirstSegmentMustBeIEdmEntityContainer);
+            }
+
+            this.targetSegments = targetSegments;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmAnnotationsTarget"/> class.
+        /// </summary>
+        /// <param name="targetSegments">Target segments.</param>
+        public EdmAnnotationsTarget(params IEdmElement[] targetSegments)
+            :this((IEnumerable<IEdmElement>)targetSegments)
+        {
+        }
+
+        /// <summary>
+        /// Gets the target segments as Edm elements.
+        /// </summary>
+        public IEnumerable<IEdmElement> TargetSegments

```

Comment:
```suggestion
        public IEnumerable<IEdmElement> Segments
```

Reply:
Or should it be `Elements` not to be confused with `Segments` of the `ODataPathSegment` variety?

Reply:
Or maybe `Segments` is not that off-the-mark https://docs.oasis-open.org/odata/odata/v4.0/cs02/part3-csdl/odata-v4.0-cs02-part3-csdl.html#_Toc372793559